Resurrect SUMO mobile tests

bobsilverberg opened the pull request:
This fixes the 5 mobile tests so they can now pass. I also improved/cleaned up the code a bit, as well as removing UnittestZero and using native `assert` for all the assertions.

Adhoc passed at http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/sumo.mobile.adhoc/2/

@m8ttyB | @rbillings | @davehunt r?

bobsilverberg commented:
Adhoc also passed against prod: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/sumo.mobile.adhoc/3/

davehunt commented on pages/mobile/search.py:
```diff
@@ -12,19 +12,23 @@
-        return [self.SearchResult(self.testsetup, web_element)
-                for web_element in self.selenium.find_elements(*self._results_locator)]
+        return [self.SearchResult(self.testsetup, element)
+                for element in self.selenium.find_elements(*self._results_locator)]
```
We use `element` here but `web_element` in `Base.menu_items`, let's be consistent (I prefer `element`)

bobsilverberg replied:
Agreed and done.

davehunt commented on tests/mobile/test_homepage.py:
```diff
@@ -6,31 +6,29 @@
     def test_the_expandable_header_menu(self, mozwebqa):
+        expected_menu_items = [u'Home', u'Ask a question', u'Support Forum',
+                               u'Help other users', u'Switch to desktop site',
+                               u'Sign in', u'Switch language']
```
Could you have each item on its own line for readability?

bobsilverberg replied:
Done

davehunt commented on pages/mobile/base.py:
```diff
@@ -11,36 +11,47 @@
-        def click_header_menu(self):
-            self.selenium.find_element(*self._menu_button_locator).click()
+    def click_menu_item(self, item_name):
+        if not self.is_menu_exposed:
```
I can't see anywhere this is being called. Can we remove it? I'm not a fan of having this kind of logic in the page objects. We should fail if the menu is not visible when we're expecting it to be.

bobsilverberg replied:
Yeah, I've removed that now. I thought I needed it as I was going through the tests, but it turned out that it wasn't used.

bobsilverberg commented:
@davehunt I have addressed your comments in a new commit. New adhoc against staging running at http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/sumo.mobile.adhoc/4/

Thanks for the review!

davehunt commented:
A minor nit, feel free to ignore. I see the adhoc failed but a rerun passed. Note that I haven't run the tests locally, but with your testing and adhoc I feel confident.
